### PR TITLE
The HTU21 sensor bring up

### PR DESCRIPTION
**Pull request by vadimol**
1) Added sensor directorty to dev
2) Added htu21 sensor implementation

### CODE REVIEW

File: dev/sensor/htu21/export/dev/sensor/htu21.hpp [hunk 1]
```diff
@@ -0,0 +1,181 @@
+#ifndef __HTU21_HPP__
```
Comment (forGGe) [inline dev/sensor/htu21/export/dev/sensor/htu21.hpp]: Let's reflect component path in define name. I.e.:
__DEV_SENSOR_HTU21_HPP__

Comment (forGGe) [inline dev/sensor/htu21/export/dev/sensor/htu21.hpp]: Add file header in doxygen style

File: dev/sensor/htu21/export/dev/sensor/htu21.hpp [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+#define __HTU21_HPP__
+
+namespace sensor {
+
+template <class i2c_dev>
```
Comment (forGGe) [inline dev/sensor/htu21/export/dev/sensor/htu21.hpp]: Add brief description of a class in doxygen style

File: dev/sensor/htu21/export/dev/sensor/htu21.hpp [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    //! \brief Reads raw relative humidity sample from sensor.
+    //! \details Sample should be processed to receive physical value.
+    //! \retval Raw relative humidity sample.
+    //!
+    static uint16_t get_sample_humidity();
+
+    //! \brief Reads sample from sensor and converts to a physical value.
+    //! \retval Temperature in C degree.
+    //!
+    static float get_temperature();
```
Comment (forGGe) [inline dev/sensor/htu21/export/dev/sensor/htu21.hpp]: That's dangerous. Not all hardaware have hardware floating point unit, and not all HW works fast. Let's use an integer, interpreted as real number. Say, value of 12345 will mean 12.345 degree in Celsius.

File: dev/sensor/htu21/export/dev/sensor/htu21.hpp [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+
+    //! \brief Reads sample from sensor and converts to a physical value.
+    //! \retval Relative humidity in %.
+    //!
+    static float get_humidity();
+
+    //! Sensor I2C address. Cannot be changed.
+    static constexpr uint8_t i2_addr = 0x80;
```
Comment (forGGe) [inline dev/sensor/htu21/export/dev/sensor/htu21.hpp]: Wondering if it worth keep it public

File: dev/sensor/htu21/export/dev/sensor/htu21.hpp [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+    //! \brief Reads sensor user register.
+    //! \details Allows to get useful information about sensor state
+    //! \retval User register content.
+    //!
+    static uint8_t read_user_register();
+
+    //! \brief Writes to the sensor user register.
+    //! \details Allows to configure sensor resolution.
+    //!
+    static void write_user_register(uint8_t value);
```
Comment (forGGe) [inline dev/sensor/htu21/export/dev/sensor/htu21.hpp]: Let's discuss a bit more this.

File: dev/sensor/htu21d/export/dev/sensor/htu21d.hpp [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+        break;
+    default:
+        break;
+    }
+
+    write_user_register(user_reg);
+}
+
+template <class i2c_dev>
+typename htu21d<i2c_dev>::resolution_t htu21d<i2c_dev>::get_resolution_mode()
```
Comment (vadimol) [inline dev/sensor/htu21d/export/dev/sensor/htu21d.hpp]: Not quite sure if it is fine, Max, what do you think?

**Comment by vadimol** [comment]
addressed review comments

File: dev/sensor/htu21d/export/dev/sensor/htu21d.hpp [hunk 7]
```diff
@@ -0,0 +1,427 @@
...
+    ecl::err rc = get_sample_temperature(sample);
+    if (rc != ecl::err::ok) {
+        return rc;
+    }
+
+    // clear last 2 bits according to RM,
+    // since they contain status information
+    sample &= ~3;
+
+    value = -46850 + ((175000 * static_cast<long long>(sample)) / (1 << 16));
```
Comment (forGGe) [inline dev/sensor/htu21d/export/dev/sensor/htu21d.hpp]: Use uint64_t for better readability.

File: dev/sensor/htu21d/export/dev/sensor/htu21d.hpp [hunk 8]
```diff
@@ -0,0 +1,427 @@
+//!
+//! \file
+//! \brief Driver for HTU21D digital humidity sensor with temperature output
+//!  See, http://www.meas-spec.com/downloads/HTU21D.pdf
+//!
+
+#ifndef __DEV_SENSOR_HTU21D_HPP__
+#define __DEV_SENSOR_HTU21D_HPP__
+
+namespace sensor {
```
Comment (forGGe) [inline dev/sensor/htu21d/export/dev/sensor/htu21d.hpp]: Add another namespace above 'sensor': 'ecl'

File: dev/sensor/htu21d/CMakeLists.txt [hunk 9]
```diff
@@ -0,0 +1,2 @@
+add_library(htu21d INTERFACE)
```
Comment (forGGe) [inline dev/sensor/htu21d/CMakeLists.txt]: Lacks dependency on 'types' library

File: dev/sensor/htu21d/export/dev/sensor/htu21d.hpp [hunk 10]
```diff
@@ -0,0 +1,427 @@
+//!
+//! \file
+//! \brief Driver for HTU21D digital humidity sensor with temperature output
+//!  See, http://www.meas-spec.com/downloads/HTU21D.pdf
+//!
+
```
Comment (forGGe) [inline dev/sensor/htu21d/export/dev/sensor/htu21d.hpp]: `#include <ecl/err.hpp>`

**Comment by forGGe** [comment]
Few comments, looks good otherwise.

File: dev/sensor/htu21d/CMakeLists.txt (hunk 9 above)
Comment (vadimol) [inline dev/sensor/htu21d/CMakeLists.txt]: could you put more details regarding this, please

Comment (forGGe) [inline dev/sensor/htu21d/CMakeLists.txt]: Specifically, you must specify that this library depends on `types` library which contains `enum class err`:
`target_link_libraries(htu21d INTERFACE types)

File: dev/sensor/htu21d/export/dev/sensor/htu21d.hpp (hunk 10 above)
Comment (vadimol) [inline dev/sensor/htu21d/export/dev/sensor/htu21d.hpp]: done

File: dev/sensor/htu21d/export/dev/sensor/htu21d.hpp (hunk 7 above)
Comment (vadimol) [inline dev/sensor/htu21d/export/dev/sensor/htu21d.hpp]: done

File: dev/sensor/htu21d/export/dev/sensor/htu21d.hpp (hunk 8 above)
Comment (vadimol) [inline dev/sensor/htu21d/export/dev/sensor/htu21d.hpp]: done

**Comment by forGGe** [comment]
Looks good

**Pull request merged**